File: Practice/04/Python/4th.py
a = int(input())
b = int(input())
a, b = b, a
print('число а теперь равно - ', a, 'число b теперь равно - ',b)

tmp = a
a = b
b = tmp
print('число а снова равно - ', a, 'число b снова равно - ',b)

File: Practice/04/C++/4th/4th/4th.cpp

#include <iostream>
using namespace std;
int main() {
    setlocale(LC_ALL, "Russian");
    int a, b;
    cout << "введите 2 переменные, сначала вводится переменная а, затем переменная b" << '\n';
    cin >> a >> b;
    // без доп. переменной
    swap(a, b);
    cout << "переменная а - " << a << "  переменная b - " << b;

    // с доп переменной
    int tmp;
    tmp = a;
    a = b;
    b = tmp;
    cout << "переменная а - " << a << "  переменная b - " << b;
    // переменные меняются значениями 2 раза, поэтому в конце переменная а=а, b=b
    return 0;

}
